src/set_theory/cardinal/finite.lean

```diff
@@ -34,6 +34,64 @@ lemma card_eq_fintype_card [fintype α] : nat.card α = fintype.card α := mk_to
 @[simp]
 lemma card_eq_zero_of_infinite [infinite α] : nat.card α = 0 := mk_to_nat_of_infinite
 
+lemma card_eq_of_equiv (f : α ≃ β) : nat.card α = nat.card β :=
+begin
+  have : ulift α ≃ ulift β := (equiv.ulift.trans f).trans equiv.ulift.symm,
+  have := cardinal.to_nat_congr this,
+  simpa using this,
+end
+
+lemma card_eq_of_bijective (f : α → β) (hf : function.bijective f) : nat.card α = nat.card β :=
+card_eq_of_equiv (equiv.of_bijective f hf)
+
+lemma card_eq_of_equiv_fin {α : Type*} {n : ℕ}
+  (f : α ≃ fin n) : nat.card α = n :=
+by simpa using card_eq_of_equiv f
+
+/-- If the cardinality is positive, that means it is a finite type, so there is
+an equivalence between `α` and `fin (nat.card α)`. See also `finite.equiv_fin`. -/
+def equiv_fin_of_card_pos {α : Type*} (h : 0 < nat.card α) :
```

Comment:
Recently, we migrated many lemmas to assume a seemingly weaker `n ≠ 0` and prove a seemingly stronger `0 < n`. Could you please assume `nat.card α ≠ 0` here?